docs/journal/01-cold-storage.md

```diff
@@ -1,74 +1,90 @@
 
-- **Connectivity**. Many simple devices just emit signals that indicate status and don't necessarily send out HTTP packets. Additionally, those devices that can create HTTP messages are unlikely to all use the same schema. It may be necessary to incorporate local field gateways to perform message translation and formatting.
+> We briefly considered whether we could use virtual machines running an open source web server, optimized specifically for this task. However, that means taking responsibility for maintaining and patching the VMs and the web server software. There are also security concerns with this approach &mdash; it's not easy to harden a custom web server.
 
-> ![Markus](media/PersonaMarkus.png) 
-"Many devices are designed to be simple; they output signals indicating events that have occurred, and they can receive signals that indicate commands to be performed. They might not be capable of translating this data into different formats for integration into an IoT solution; this process typically requires some auxiliary hardware and/or software."
+- **Connectivity**. Many simple devices can't send HTTP messages, and devices that *can* send HTTP messages are unlikely to use the same schema. Therefore, a field gateway may be needed to translate device messages.
 
-![cloud service solution based on web roles](media/01-cold-storage/physical-architecture-cloud-service.png)
+> Many devices are designed to be simple; they output signals indicating events, and can receive signals for commands to be performed. They might not be capable of translating this data into different formats for integration into an IoT solution; this process typically requires some auxiliary hardware and/or software.
 
-- **Resilience and reliability.** If an instance of the service fails, then any messages that it is processing might be lost.
+- **Resilience and reliability.** If an instance of the service fails, then any messages it is processing might be lost.
 
-- **Scalability.** Event data might need to be handled in multiple ways, such as copying to cold storage, performing warm analysis and aggregation, or for raising alerts. If all of these tasks become the responsibility of the cloud service it might not be capable of handling this work together with the large ingress of event information.
+- **Scalability.** The HTTP service handles both ingestion and processing. That's feasible for the cold-storage scenario, but what happens when we add more processing tasks, such as warm storage, event aggregation, or critical alerts? Can the service still handle the load *and* process events quickly enough? A big advantage of the other solutions we explored is the use of a publisher/consumer model, separating ingestion from processing.      
```

Comment:
What do you think about pointing out how a surge in traffic might cause ingestion to fail because the downstream processing can't keep up? This is a problem when the two are tied together.